test/framework/src/main/java/org/elasticsearch/test/disruption/DisruptableMockTransport.java

```diff
@@ -87,8 +87,14 @@ public TransportService createTransportService(Settings settings, ThreadPool thr
 
     @Override
     public Releasable openConnection(DiscoveryNode node, ConnectionProfile profile, ActionListener<Connection> listener) {
-        final Optional<DisruptableMockTransport> matchingTransport = getDisruptableMockTransport(node.getAddress());
-        if (matchingTransport.isPresent()) {
+        final Optional<DisruptableMockTransport> optionalMatchingTransport = getDisruptableMockTransport(node.getAddress());
+        if (optionalMatchingTransport.isPresent()) {
+            final DisruptableMockTransport matchingTransport = optionalMatchingTransport.get();
+            final ConnectionStatus connectionStatus = getConnectionStatus(matchingTransport.getLocalNode());
+            if (connectionStatus != ConnectionStatus.CONNECTED) {
```

Comment:
I wonder if black holing should just not call the listener here. Right now, it's a bit tricky because this is currently a blocking call. Once we move to async, this will become easier however.

Reply:
Agreed, there's no good solution for blackhole connections here yet.